azurerm/resourceid.go

```diff
@@ -101,23 +101,29 @@ func parseAzureResourceID(id string) (*ResourceID, error) {
 	return idObj, nil
 }
 
-func composeAzureResourceID(idObj *ResourceID) (id string, err error) {
-	if idObj.SubscriptionID == "" || idObj.ResourceGroup == "" {
-		return "", fmt.Errorf("SubscriptionID and ResourceGroup cannot be empty")
+func composeAzureResourceID(subscriptionID string, resourceGroupName string, provider string, path []string) (id string, err error) {
```

Comment:
is the intention that we wrap this method for each type of resource? There's also an [issue open on the SDK](https://github.com/Azure/azure-sdk-for-go/issues/442) to add support for this